include/seastar/core/abort_source.hh

```diff
@@ -104,13 +104,14 @@ private:
     std::exception_ptr _ex;
 
     void do_request_abort(std::optional<std::exception_ptr> ex) noexcept {
-        assert(_subscriptions);
-        _ex = ex.value_or(get_default_exception());
-        auto subs = std::exchange(_subscriptions, std::nullopt);
-        while (!subs->empty()) {
-            subscription& s = subs->front();
-            s.unlink();
-            s.on_abort(ex);
+        if (_subscriptions) {
+            _ex = ex.value_or(get_default_exception());
```

Comment:
Should _subscriptions be an std::optional? A list can be empty, so why those two levels of optionality?

Reply:
I guess the optional was used before to mark a yet-to-be-aborted abort_source, but now that we're allowing multiple aborts, it doesn't serve any purpose.

Reply:
Right, I too think it's not needed any more

Reply:
right